### ISSUE TITLE
Support for Marker Image URLs

### ISSUE THREAD

**Issue by burkeholland** [Yermo/nativescript-mapbox#20]
Currently, the plugin only loads local resources as marker images. It would be nice if we could also specify a URL to use for the marker.

**Comment by EddyVerbruggen** [comment]
Markers need to be downloaded before they can be used, and the plugin already supports adding markers from local files so it's doable currently but it would be user friendly to have the plugin handle the download process.

A tricky thing here is when multiple markers need to be added with different "remote" images things get messed up as downloads are async.

**Comment by EddyVerbruggen** [comment]
Note to self: on iOS downloading marker images likely requires ATS exceptions to be added to the plist

**Comment by EddyVerbruggen** [comment]
So you can now do one of these on both iOS and Android:

``` js
icon: 'http://www.bme.be/wp-content/uploads/2014/04/marker.png'
icon: 'res://icon'
iconPath: 'res/markers/home_marker.png'
```

I've implemented a high performance algorithm that fetches and caches markers from urls. It's probably not perfect but for a first attempt I'm pretty happy.

This map uses images configured in various ways:

![simulator screen shot 16 nov 2016 22 40 50](https://cloud.githubusercontent.com/assets/1426370/20366959/c08d16c8-ac4d-11e6-9186-86d6c707defb.png)

Available from GitHub now, on npm when 2.3.0 is published.
